plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -47,17 +47,36 @@ public open class DefaultPageCreator(
     protected val separateInheritedMembers: Boolean =
         configuration?.separateInheritedMembers ?: DokkaBaseConfiguration.separateInheritedMembersDefault
 
-    public open fun pageForModule(m: DModule): ModulePageNode =
-        ModulePageNode(m.name.ifEmpty { "<root>" }, contentForModule(m), listOf(m), m.packages.map(::pageForPackage))
+    public open fun pageForModule(m: DModule): ModulePageNode {
+        val hasTypes = m.packages.any {
+            it.classlikes.isNotEmpty() || it.typealiases.isNotEmpty()
+        }
+        val packagePages = m.packages.map(::pageForPackage)
+        return ModulePageNode(
+            name = m.name.ifEmpty { "<root>" },
+            content = contentForModule(m),
+            documentables = listOf(m),
+            children = when {
+                hasTypes -> packagePages + AllTypesPageNode(
+                    name = "All Types",
```

Comment:
It depends, I think. F.e. on `PageNode` and `ContentNode` there are functions named `modified` which allows to modify the node via extension. So may be modifying name should be allowed.
But while for now we hide this functionality under a flag, we can ignore possibility to modify
I will make name and children defined in `AllTypesPageNode`

Reply:
`modified` returns a modified copy of a page.
Anyway, users can define their own `PageNode` with `content` from `AllTypesPageNode`.

>  But while for now we hide this functionality under a flag, we can ignore possibility to modify

I would even make `AllTypesPageNode` internal since it is not stable.

Reply:
made internal in https://github.com/Kotlin/dokka/pull/3267/commits/043e632a7fd4ce7c0fd8227565ce9eff7968bbea